aiida/cmdline/params/options/interactive.py

```diff
@@ -166,42 +176,40 @@ def after_callback(self, ctx, param, value):
     def prompt_callback(self, ctx, param, value):
         """decide wether to initiate the prompt_loop or not"""
 
-        # a value was given
+        # a value was given on the command line: then  just go with validation
         if value is not None:
             return self.after_callback(ctx, param, value)
 
-        # parameter is not reqired anyway
-        if not self.is_required(ctx):
-            return self.after_callback(ctx, param, value)
+        # The same if the user specified --non-interactive
+        if noninteractive(ctx):
+            # Check if it is required
 
-        # help option was passed on the cmdline
-        if ctx.params.get('help'):
-            return self.after_callback(ctx, param, value)
```

Comment:
Is it tested somewhere that removing this never causes an InteractiveOption to enter the prompt loop before `--help` gets processed?

Reply:
If it wasn't tested before, then no. It wasn't clear to me the purpose of that logic. Please feel free to add a test for what you have in mind.